### PR TITLE
Ci web demo

### LINKED ISSUE #23: Establish Web Build
_I'm opening some issues for possible next tasks. Just let me know what you think about them._

It'd probably be good to setup the web build.

It should be easy and I've got a nice HTML file with a simple loading screen ( while WASM is downloading ), and I have a simple system that will make sure the game plays in full screen and auto-plays sound to work around weird chrome browser problems.

loading-game

### PR DESCRIPTION
**Pull request by zicklag**
Depends on #23.

Play it here: https://zicklag.github.io/punchy/demo/

Fixes #23

This will allow people to follow our progress super easily. We'll have an always up-to-date web demo.

### CODE REVIEW

**Comment by erlend-sh** [comment]
Fantastic! v0.0.2 is gonna be packed full of good stuff 💝

File: .github/workflows/web-demo.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Publish Web Demo
+
+on:
+  push:
```
[the remark this answers is not in the document]
Comment (zicklag) [reply .github/workflows/web-demo.yml:4]: We could do it on a tag. If we have 2000 minutes per month and it takes ~11 minutes to build then we could only run ~180 builds ( and that's if we _only_ had this CI job ), so yeah, I guess for now we can just move the tag whenever we want to update the web build and that wouldn't be too bad.

[the remark this answers is not in the document]
Comment (odecay) [reply .github/workflows/web-demo.yml:4]: Probably good to have it match web-demo and every semver tag so its updated whenever we do a release as well.

**Comment by zicklag** [comment]
So to get the web build hosted for this Repo you should just have to push a `web-demo` tag and then go to the repo settings and enable GitHub pages for the root dir of the `gh-pages` branch.

Then we can set the repo URL to link to the demo.

**Comment by odecay** [comment]
Lets see we probably have to enable pages for this to work huh.

**Comment by odecay** [comment]
Okay done web build lives at https://fishfight.github.io/punchy/demo/

**Comment by zicklag** [comment]
Awesome! You want to put the link to it in the repo meta so it pops up here:

image

**Pull request merged**